inspirehep/modules/orcid/tasks.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Manage ORCID OAUTH token migration from INSPIRE legacy instance."""
+
+from __future__ import absolute_import, division, print_function
+
+from flask import current_app
+from celery import shared_task
+from celery.utils.log import get_task_logger
+from redis import StrictRedis
+from redis_lock import Lock
+from ast import literal_eval
+from invenio_oauthclient.utils import oauth_get_user, oauth_link_external_id
+from invenio_oauthclient.models import RemoteToken, User, RemoteAccount, UserIdentity
+from invenio_db import db
+from inspire_utils.record import get_value
+from invenio_oauthclient.errors import AlreadyLinkedError
+
+logger = get_task_logger(__name__)
+
+
+def legacy_orcid_tuples():
+    """Generator to fetch token data from redis.
+
+    Yields:
+        tuple: user data in the form of (orcid, token, email, name)
+    """
+    redis_url = current_app.config.get('CACHE_REDIS_URL')
+    r = StrictRedis.from_url(redis_url)
+    lock = Lock(r, 'import_legacy_orcid_tokens', expire=120, auto_renewal=True)
+    if lock.acquire(blocking=False):
+        try:
+            while r.llen('legacy_orcid_tokens'):
+                yield literal_eval(r.lrange('legacy_orcid_tokens', 0, 1)[0])
```

Comment:
As discussed IRL, that's unfortunate due to my wrong assumption on Redis serialization. Let's stick to exchange data using JSON serialized string (in this case it will be a simple JSON array with positional values).